btell_main/forms/register.py

```diff
@@ -0,0 +1,150 @@
+# from btell_main.models import Profile
+from django.views.generic.edit import FormView
+from django.shortcuts import redirect
+from django.urls import reverse_lazy
+from django import forms
+from django.core.validators import validate_email
+from django.contrib.auth import authenticate, login
+# from django.contrib.auth import views as auth_views
+from django.contrib.auth import models as auth_models
+from django.contrib.auth import forms as auth_forms
+
+# from datetime import datetime, timezone
+
+
+class Register_User(auth_forms.UserCreationForm):
+    email = forms.EmailField(required=True, label='Email')
+    accepted_cookies = forms.BooleanField(required=True, label='Cookies')
+    accepted_tos = forms.BooleanField(required=True, label='ToS')
+
+    class Meta:
+        model = auth_models.User
+        fields = (
+            'username',
+            'password1',
+            'password2',
+            'email',
+            'accepted_cookies',
+            'accepted_tos',
+        )
+
+    def clean(self):
+        data = self.cleaned_data
+
+        u1 = ""
+        if 'username' in data:
+            u1 = str(data['username'])
+
+        if u1 != "":
+            if len(u1) > 100:  # TODO Decide max username lenght
+                self.add_error('username', "The name is too long. Max 100 marks.")
+
+            u_check = auth_models.User.objects.filter(username=u1)
+            if u_check.exists():
+                self.add_error('username', "Such Player already exists.")
+
+        e1 = ""
+        if 'email' in data:
+            e1 = data['email']
+
+        if e1 != "":
+            if validate_email(e1) is not None:
+                self.add_error('email', "It's not in e-mail format.")
+
+            if len(e1) > 100:  # TODO Decide max e-mail lenght
+                self.add_error('email', "E-mail is too long. Max 100 marks.")
+
+            e_check = auth_models.User.objects.filter(email=e1)  # TODO If you want it
+            if e_check.exists():
+                self.add_error('email', "Email is already assigned to some account.")
+
+        p1 = ""
+        if 'password1' in data:
+            p1 = data['password1']
+
+        # if p1 != "":  # TODO Pick any password requirements you want
```

Comment:
We can use https://docs.djangoproject.com/en/4.2/topics/auth/passwords/#integrating-validation to automatically validate password based on `settings.py`. There is also `password_validators_help_texts()` which returns a list of helper texts to explain the requirements.

Reply:
We can do it later on, once we know what kind of password validators do we want; lenght, capitals, numbers, special.

Reply:
That's just it - using these two functions will mean it obeys whatever is set in the settings, and we don't have to keep changing this.

Reply:
I agree we can put it in later, though, yes.